### PR TITLE
Test bugzilla

### PR DESCRIPTION
**Pull request by mjagelka**
Add unit tests for bugzilla.py and minor changes in original code.

### CODE REVIEW

File: ticketutil/bugzilla.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ class BugzillaTicket(ticket.Ticket):
     """
     A BZ Ticket object. Contains BZ-specific methods for working with tickets.
     """
-    def __init__(self, url, project, auth=None, ticket_id=None):
+    def __init__(self, url, project, auth='kerberos', ticket_id=None):
```
Comment (dmranck) [inline ticketutil/bugzilla.py]: This should go back to auth=None. BugzillaTicket supports HTTP Basic auth, Kerberos auth, and API Key auth, but we don't want to necessarily set a default of 'kerberos'.

File: tests/test_bugzilla.py [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+        self.assertEqual(None, ticket._generate_ticket_url())
+        self.assertEqual(ticket.request_result, SUCCESS_RESULT)
+
+    @patch.object(ticketutil.ticket.Ticket, '_create_requests_session')
+    def test_create_requests_session_kerberos_auth(self, mock_session):
+        mock_session.return_value = FakeSession()
+        with patch.object(bugzilla.BugzillaTicket, '_create_requests_session'):
+            ticket = bugzilla.BugzillaTicket(URL, PROJECT)
+        t = ticket._create_requests_session()
+        mock_session.assert_called_once()
```
Comment (dmranck) [inline tests/test_bugzilla.py]: This line is giving me an AttributeError when I run these unit tests, failing this test case.

File: tests/test_bugzilla.py [hunk 3]
```diff
@@ -0,0 +1,498 @@
...
+        mock_session.return_value = FakeSession()
+        ticket = bugzilla.BugzillaTicket(URL, PROJECT)
+        self.assertEqual(None, ticket._generate_ticket_url())
+        self.assertEqual(ticket.request_result, SUCCESS_RESULT)
+
+    @patch.object(ticketutil.ticket.Ticket, '_create_requests_session')
+    def test_create_requests_session_kerberos_auth(self, mock_session):
+        mock_session.return_value = FakeSession()
+        with patch.object(bugzilla.BugzillaTicket, '_create_requests_session'):
+            ticket = bugzilla.BugzillaTicket(URL, PROJECT)
```
Comment (dmranck) [inline tests/test_bugzilla.py]: If you make the change in bugzilla.py below, you'll probably need to add the following to this line:

`auth='kerberos'`

File: ticketutil/bugzilla.py (hunk 1 above)
Comment (mjagelka) [reply ticketutil/bugzilla.py]: I forgot to comment on this. If `auth=None` it breaks on l.67 `elif 'api_key' in self.auth:`. So there is an option to set `auth='kerberos'` as default or somehow deal with auth being None through some additional condition (if not self.auth: logger.error()... elif 'api_key' in self.auth: ... ). I chose the simplest way.

File: tests/test_bugzilla.py (hunk 2 above)
Comment (mjagelka) [reply tests/test_bugzilla.py]: This should be a problem of python version. `assert_called_once` works for python 3.6 and higher.

File: tests/test_bugzilla.py (hunk 3 above)
Comment (mjagelka) [reply tests/test_bugzilla.py]: I am aware of that. I will wait for your response to the comment above.

File: ticketutil/bugzilla.py (hunk 1 above)
Comment (dmranck) [reply ticketutil/bugzilla.py]: Hmm, good point... Ok, I agree. Keep it as `auth='kerberos'`

**Comment by dmranck** [comment]
Addressed your responses to my comments above. Everything's running well on Python 3.6. We can merge this if you're ready.

**Comment by mjagelka** [comment]
I am ready.

**Pull request merged**